server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -131,6 +134,16 @@ void setUpdateFrequency(TimeValue updateFrequency) {
         this.updateFrequency = updateFrequency;
     }
 
+    /**
+     * This can be provided by plugins, which are initialised long after the ClusterInfoService is created
```

Comment:
Why does this need to be provided by plugins, shouldn't it be backed by JvmInfo/JvmStats?

Reply:
The heap usage supplier supplies estimated heap usage for the whole cluster, the single implementation we have of this exists only in Serverless at the moment, we're using the heap usage estimates provided by `MemoryMetricsService` which are aggregated on the master using transport layer messages.

The heap usage in this case is not the JVM-provided heap usage but the estimate for the shards that are allocated to a node (i.e. not "real" heap usage). The estimate is the same one we use for autoscaling.